File: Android/app/src/androidTest/java/jp/co/riso/android/text/AlphaNumericFilterTest.java
package jp.co.riso.android.text;

import jp.co.riso.smartdeviceapp.SmartDeviceApp;
import jp.co.riso.smartdeviceapp.view.MainActivity;
import android.test.ActivityInstrumentationTestCase2;
import android.text.InputFilter;
import android.widget.EditText;

public class AlphaNumericFilterTest extends  ActivityInstrumentationTestCase2<MainActivity> {
    private static final String INVALID_INPUT_TEXT = "!";
    private static final String VALID_INPUT_TEXT = "a";
    private static final String KANJI_TEXT_ONLY = "この言葉は漢字です。";
    private static final String KANJI_WITH_NUMBERS = "この言葉は漢字です。12345";
    private static final String KANJI_WITH_LETTERS = "ABCDEこの言葉は漢字です。";
    private static final String KANJI_WITH_ALPHANUM = "この言葉12345は漢字です。ABCDE";
    
    public AlphaNumericFilterTest() {
        super(MainActivity.class);
    }

    protected void setUp() throws Exception {
        super.setUp();
    }

    protected void tearDown() throws Exception {
        super.tearDown();
    }
    
    // ================================================================================
    // Tests - constructors
    // ================================================================================

    public void testConstructor() {
        AlphaNumericFilter filter = new AlphaNumericFilter();
        assertNotNull(filter);
    }
    
    // ================================================================================
    // Tests - filter
    // ================================================================================

    public void testFilter() {
        EditText editText = new EditText(SmartDeviceApp.Companion.getAppContext());
        editText.setFilters(new InputFilter[] {
                new InputFilter.LengthFilter(8),
                new AlphaNumericFilter()
        });
        editText.setText(INVALID_INPUT_TEXT);
        assertEquals("", editText.getText().toString());
        editText.setText(VALID_INPUT_TEXT);
        assertEquals(VALID_INPUT_TEXT, editText.getText().toString());
    }
    
    public void testFilter_Kanji() {
        EditText editText = new EditText(SmartDeviceApp.Companion.getAppContext());
        editText.setFilters(new InputFilter[] {
                new AlphaNumericFilter()
        });
        editText.setText(KANJI_TEXT_ONLY);
        assertEquals("", editText.getText().toString());
        editText.setText(KANJI_WITH_NUMBERS);
        assertEquals("12345", editText.getText().toString());
        editText.setText(KANJI_WITH_LETTERS);
        assertEquals("ABCDE", editText.getText().toString());
        editText.setText(KANJI_WITH_ALPHANUM);
        assertEquals("12345ABCDE", editText.getText().toString());
    }
    
}

File: Android/app/src/androidTest/kotlin/jp/co/riso/android/text/AlphaNumericFilterTest.kt
package jp.co.riso.android.text

import android.text.InputFilter
import android.text.InputFilter.LengthFilter
import android.widget.EditText
import jp.co.riso.smartdeviceapp.SmartDeviceApp
import junit.framework.TestCase
import org.junit.Test

class AlphaNumericFilterTest {

    // ================================================================================
    // Tests - constructors
    // ================================================================================
    @Test
    fun testConstructor() {
        val filter = AlphaNumericFilter()
        TestCase.assertNotNull(filter)
    }

    // ================================================================================
    // Tests - filter
    // ================================================================================
    @Test
    fun testFilter() {
        val editText = EditText(SmartDeviceApp.appContext!!)
        editText.filters = arrayOf(
            LengthFilter(8),
            AlphaNumericFilter()
        )
        editText.setText(INVALID_INPUT_TEXT)
        TestCase.assertEquals("", editText.text.toString())
        editText.setText(VALID_INPUT_TEXT)
        TestCase.assertEquals(VALID_INPUT_TEXT, editText.text.toString())
    }

    @Test
    fun testFilter_Kanji() {
        val editText = EditText(SmartDeviceApp.appContext!!)
        editText.filters = arrayOf<InputFilter>(
            AlphaNumericFilter()
        )
        editText.setText(KANJI_TEXT_ONLY)
        TestCase.assertEquals("", editText.text.toString())
        editText.setText(KANJI_WITH_NUMBERS)
        TestCase.assertEquals("12345", editText.text.toString())
        editText.setText(KANJI_WITH_LETTERS)
        TestCase.assertEquals("ABCDE", editText.text.toString())
        editText.setText(KANJI_WITH_ALPHANUM)
        TestCase.assertEquals("12345ABCDE", editText.text.toString())
    }

    companion object {
        private const val INVALID_INPUT_TEXT = "!"
        private const val VALID_INPUT_TEXT = "a"
        private const val KANJI_TEXT_ONLY = "この言葉は漢字です。"
        private const val KANJI_WITH_NUMBERS = "この言葉は漢字です。12345"
        private const val KANJI_WITH_LETTERS = "ABCDEこの言葉は漢字です。"
        private const val KANJI_WITH_ALPHANUM = "この言葉12345は漢字です。ABCDE"
    }
}